src/Features/VisualBasic/Portable/ExtractInterface/VisualBasicExtractInterfaceService.vb

```diff
@@ -88,46 +71,39 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.ExtractInterface
             Return fullDisplayName
         End Function
 
-        Private Function UpdateTypeWithImplementsClause(
-            solutionWithInterfaceDocument As Solution,
-            invocationLocationDocument As DocumentId,
-            typeNodeAnnotation As SyntaxAnnotation,
-            extractedInterfaceSymbol As INamedTypeSymbol,
-            docToRootMap As Dictionary(Of DocumentId, CompilationUnitSyntax),
-            cancellationToken As CancellationToken) As String
+        Private Function GetUpdatedImplementsClause(implementsClause As ImplementsClauseSyntax, qualifiedName As QualifiedNameSyntax) As ImplementsClauseSyntax
+            If implementsClause IsNot Nothing Then
+                Return implementsClause.AddInterfaceMembers(qualifiedName).WithAdditionalAnnotations(Formatter.Annotation)
+            Else
+                Return SyntaxFactory.ImplementsClause(qualifiedName).WithAdditionalAnnotations(Formatter.Annotation)
+            End If
+        End Function
+
+        Private Function CreateFinalSolution(solutionWithInterfaceDocument As Solution, documentIds As IEnumerable(Of DocumentId), docToRootMap As Dictionary(Of DocumentId, CompilationUnitSyntax)) As Solution
+            Dim finalSolution = solutionWithInterfaceDocument
+
+            For Each docId In documentIds
+                finalSolution = finalSolution.WithDocumentSyntaxRoot(docId, docToRootMap(docId), PreservationMode.PreserveIdentity)
+            Next
+
+            Return finalSolution
+        End Function
+
+        Friend Overrides Function ShouldIncludeAccessibilityModifier(typeNode As SyntaxNode) As Boolean
+            Dim typeDeclaration = DirectCast(typeNode, TypeBlockSyntax)
+            Return typeDeclaration.GetModifiers().Any(Function(m) SyntaxFacts.IsAccessibilityModifier(m.Kind()))
+        End Function
 
-            Dim documentWithTypeNode = solutionWithInterfaceDocument.GetDocument(invocationLocationDocument)
-            Dim typeDeclaration = documentWithTypeNode.GetSyntaxRootSynchronously(cancellationToken).GetAnnotatedNodes(Of TypeBlockSyntax)(typeNodeAnnotation).Single()
+        Friend Overrides Function UpdateMembersWithExplicitImplementations(unformattedSolution As Solution, documentIds As IReadOnlyList(Of DocumentId), extractedInterfaceSymbol As INamedTypeSymbol, typeToExtractFrom As INamedTypeSymbol, includedMembers As IEnumerable(Of ISymbol), symbolToDeclarationAnnotationMap As Dictionary(Of ISymbol, SyntaxAnnotation), cancellationToken As CancellationToken) As Solution
+            Dim docToRootMap = New Dictionary(Of DocumentId, CompilationUnitSyntax)
 
             Dim implementedInterfaceStatementSyntax = If(extractedInterfaceSymbol.TypeParameters.Any(),
```

Comment:
for thing like [Property], does extractedInterfaceSymbol.Name include escaping char [] or just Property? do we have a test for it?

we have this extension method - http://source.roslyn.io/#Microsoft.CodeAnalysis.CSharp.Workspaces/Extensions/StringExtensions.cs,12 or 
http://source.roslyn.io/#Microsoft.CodeAnalysis.VisualBasic.Workspaces/Extensions/StringExtensions.vb,12

shouldn't it use those to correctly escape and stuff?